### ISSUE TITLE
[Enhancement] AutoSync docs

### ISSUE THREAD

**Issue by yqlbu** [daeuniverse/dae-docs#34]
## Context

Currently, `dae-docs` is ONLY capable of synchronizing `example.dae` on the fly from [dae](https://github.com/daeuniverse/dae). In other words, whenever `example.dae` is changed in `dae`, @daebot will trigger [Synchronize document](https://github.com/daeuniverse/dae/actions/workflows/sync-docs.yml) workflow and create a pull request in this repo.

However, the `full-docs-sync` functionality has NOT been fully implemented. The purpose of this issue is to fulfill the following business requirements:

- Sync docs-related files end with `*.md` on the fly so that `dae-docs` can always be up-to-date to the upstream.

## Scope

- [ ] Draft a technical solution design to address the above need
- [ ] Refine the current sync-docs workflow in dae to handle all sync scenarios
- [ ] Write documentation preview link in the draft PR raised in `dae` so that the author of the PR can preview documentation changes before proceeding to merge the PR.

## Expected Outcomes

- Whenever a new PR (PR-A) is raised in `dae`, the `sync-docs` action will be triggered and create a PR (PR-B) in this repo (dae-docs) with all docs-related changes proposed in PR-A. Ideally, every single docs-related commit has to be taken into consideration.
- A preview version of the docs, including all new proposed changes, should be deployed automatically with existing workflow in `dae-docs`; once the preview deployment is ready, post the URL in `dae` as a comment in PR-A.

## Implementation

TBD.
